Add configuration details to the README (#100)
bbc/nmos-common: README.md

$ sudo python setup.py install
```

## Configuration

This library provides a number of configuration options which are used by its own components and by services which depend upon the library. Configuration is provided by a JSON object within the file `/etc/nmoscommon/config.json`. The following attributes may be set in the object:

> *   **priority:** Overrides the default behaviour for interaction with a Registration API, causing it to prefer a particular instance if set to a value greater than or equal to 100. Default: 0.

Reviewer: Does this need to be emphasized? 
```suggestion
*   **priority:** Overrides the default behaviour for interaction with a Registration API, causing it to prefer a particular instance **if set to a value greater than or equal to 100**. Default: 0.
```

*   **prefer_ipv6:** A boolean parameter to switch the libary into IPv6 mode. Default: false.
*   **prefer_hostnames:** A boolean parameter to cause hostnames to be surfaced via APIs rather than IP addresses, and to cause any HTTP requests to use hostnames rather than IP addresses where possible. This is important when operating in HTTPS mode. Default: false.
*   **node_hostname:** Overrides the default fully qualified DNS name for the system. Default: null.
*   **fix_proxy:** Corrects the appearance of URLs and links (including some headers) returned by APIs when operating behind a reverse proxy. Default: "disabled".
*   **logging.level:** Sets the log level to one of "DEBUG", "INFO", "WARN", "ERROR" or "FATAL". Default: "DEBUG".
*   **logging.fileLocation:** The path to write the log file to. Default: "/var/log/nmos.log".
*   **logging.output:** An array of locations to write log messages to, which may include "file" and "stdout". Default: \["file", "stdout"\].
*   **nodefacade.NODE_REGVERSION:** Which Registration API version to look for when performing registrations from a Node API. Default: "v1.2".

> *   **oauth_mode:** Boolean indicating whether use of JSON Web Tokens and OAuth authorisation is expected. Default: false.

Reviewer: My OCD:

```suggestion
*   **oauth_mode:** A boolean parameter indicating whether use of JSON Web Tokens and OAuth authorisation is expected. Default: false.
```

*   **dns_discover:** A boolean parameter indicating whether browsing of DNS-SD via unicast DNS is enabled. Default: true.
*   **mdns_discover:** A boolean parameter indicating whether browsing of DNS-SD via multicast DNS is enabled. Default: true.

> *   **interfaces:** An array of interfaces which mDNS announcements and browsing should be performed via. Default: Not present (uses the default gateway interface).

Reviewer: Nit pick:

```suggestion
*   **interfaces:** An array containing network interfaces, via which mDNS announcements and browsing should be performed. Default: Not present (uses the default gateway interface).
```
